### PR TITLE
fix: Specify transfer precedence to match mono behavior

### PR DESCRIPTION
**Pull request by mhess-swl**

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersDecoder.java [hunk 1]
```diff
@@ -328,15 +330,33 @@ private TokenTransferList adjustingUnits(
...
             @NonNull final AccountID from,
             @NonNull final AccountID to,
             final long amount,
-            final IsApproval isApproval) {
+            final IsApproval isApproval,
+            @NonNull final TransferPrecedence precedence) {
         final var accountAmounts = new ArrayList<AccountAmount>();
-        accountAmounts.add(debit(from, amount, isApproval));
-        accountAmounts.add(credit(to, amount));
+        if (precedence == TransferPrecedence.DEBIT) {
```
Comment (lukelee-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersDecoder.java:353]: yikes.  this is scary that this matters.

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Review by tinker-michaelj — APPROVED** [review]
LGTM, thanks @mhess-swl esp for adding the unit test!

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersDecoder.java (hunk 1 above)
Comment (mhess-swl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersDecoder.java:353]: The only difference it made in the record stream was a different ordering of account amounts. This is purely to match mono service behavior; we can probably remove this pretty quickly after deployment (and I suspect we probably should, cuz agreed – this shouldn't matter)

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @mhess-swl !

**Pull request merged**